Cannot override idris-no-deps in idrisPackages for builtin packages

## Issue description

Overriding idris-no-deps in nixpkgs results in a mismatch between versions of builtin packages and the compiler.

### Steps to reproduce

With this derivation:

```
{ fetchFromGitHub, haskell, haskellPackages, idrisPackages }:
let
  idrisLatestSrc = fetchFromGitHub {
    owner = "idris-lang";
    repo = "Idris-dev";
    rev = "611564bc948029f4eb6f38154d417fd782aad789";
    sha256 = "0hy2a21vpy4p0hk01kxzlh943l55j6isdi1ivx0dx3089vd634qs";
  };

  idrisLatest = let
    haskellPackagesOverriden = haskellPackages.override {
      overrides = self: super: {
        idris = haskell.lib.dontCheck (self.callCabal2nix "idris" idrisLatestSrc {});
      };
  }; in haskellPackagesOverriden.idris;

  idrisLatestPackages = idrisPackages.override (old: {
    idris-no-deps = idrisLatest;
  });
  build-builtin-package-latest = idrisLatestPackages.build-builtin-package.override { idris = idrisLatest; };
  idrisPreludeLatest = build-builtin-package-latest "prelude" [];
in idrisLatestPackages.with-packages [ idrisPreludeLatest ]
```

I would expect to end up overriding the version of idris-no-deps in idrisPackages would change the derivations of builtin packages to be built with the supplied version of Idris. However, instead they are built with the Idris from the channel used. Leading to this error at build time (for Idris2):

```
john@nixos ~/W/s/g/e/Idris2> make
idris --build idris2.ipkg
Entering directory `./src'

/nix/store/kw6xyppzyazhjv7vymrmwjsa5zpvzgq0-idris-1.3.1/libs/prelude/Builtins.idr: openFile: does not exist (No such file or directory)

/nix/store/kw6xyppzyazhjv7vymrmwjsa5zpvzgq0-idris-1.3.1/libs/prelude/Prelude.idr: openFile: does not exist (No such file or directory)
 Can't find import Data/Vect
make: *** [Makefile:12: idris2] Error 1
```

By examining the final result of the build I can see 2 versions of idris being used:

```
john@nixos ~> nix-store -q --graph (which idris) | rg "idris-1.3.1\" ->"
"/nix/store/h08x6yzqcznmlkg6d0gp7l5rk3y2xnkq-idris-1.3.1" -> "/nix/store/kw6xyppzyazhjv7vymrmwjsa5zpvzgq0-idris-1.3.1" [color = "red"];
"/nix/store/yffixj181ya0gqrdl7fkx1qyc9fcdc17-idris-1.3.1" -> "/nix/store/kw6xyppzyazhjv7vymrmwjsa5zpvzgq0-idris-1.3.1" [color = "blue"];
"/nix/store/h08x6yzqcznmlkg6d0gp7l5rk3y2xnkq-idris-1.3.1" -> "/nix/store/yffixj181ya0gqrdl7fkx1qyc9fcdc17-idris-1.3.1" [color = "black"];
```
## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the
results.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.58, NixOS, 19.03.173088.88f069e2d8a (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(john): `"nixos-18.03-18.03.133402.cb0e20d6db9, nixpkgs-19.09pre184803.d567c486ca5, oldnixos-18.09.2574.a7e559a5504"`
 - channels(root): `"nixos-19.03.173088.88f069e2d8a"`
 - nixpkgs: `/home/john/.nix-defexpr/channels/nixpkgs`
```

--- a/pkgs/development/idris-modules/build-idris-package.nix
+++ b/pkgs/development/idris-modules/build-idris-package.nix
@@ -1,5 +1,5 @@
 # Build an idris package
-{ stdenv, lib, idrisPackages, gmp }:
+{ stdenv, lib, gmp, prelude, base, with-packages, idris }:
   { idrisDeps ? []
   , noPrelude ? false
   , noBase ? false
@@ -11,15 +11,15 @@
   }@attrs:
 let
   allIdrisDeps = idrisDeps
-    ++ lib.optional (!noPrelude) idrisPackages.prelude
-    ++ lib.optional (!noBase) idrisPackages.base;
-  idris-with-packages = idrisPackages.with-packages allIdrisDeps;
+    ++ lib.optional (!noPrelude) prelude
+    ++ lib.optional (!noBase) base;
+  idris-with-packages = with-packages allIdrisDeps;
   newAttrs = builtins.removeAttrs attrs [
     "idrisDeps" "noPrelude" "noBase"
     "name" "version" "ipkgName" "extraBuildInputs"
   ] // {
     meta = attrs.meta // {
-      platforms = attrs.meta.platforms or idrisPackages.idris.meta.platforms;
+      platforms = attrs.meta.platforms or idris.meta.platforms;
     };
   };
 in

